mne/datasets/utils.py

```diff
@@ -214,20 +218,25 @@ def has_dataset(name):
     """
     from mne.datasets._fetch import fetch_dataset
 
-    name = 'spm' if name == 'spm_face' else name
-    dataset_params = MNE_DATASETS[name]
-    dataset_params['dataset_name'] = name
-    config_key = MNE_DATASETS[name]['config_key']
+    if isinstance(name, dict):
+        dataset_name = name['dataset_name']
+        dataset_params = name
+    else:
+        dataset_name = 'spm' if name == 'spm_face' else name
+        dataset_params = MNE_DATASETS[dataset_name]
+        dataset_params['dataset_name'] = dataset_name
```

Comment:
I find myself wondering why we don't just hard-code the dataset name into those dicts in the first place. IIRC there was some reason to want them there anyway (for the downloading?)

Reply:
Yea for the downloading, we would want them there. It's up to you all. I had removed it because then it's just cleaner config file, but I can add it back in to get rid of lines like these.